File: jbake-core/src/main/java/org/jbake/template/model/TypedDocumentsExtractor.java
package org.jbake.template.model;

import org.jbake.app.ContentStore;
import org.jbake.app.DocumentList;
import org.jbake.model.DocumentModel;
import org.jbake.model.DocumentTypeUtils;
import org.jbake.template.ModelExtractor;

public class TypedDocumentsExtractor implements ModelExtractor<DocumentList<DocumentModel>> {

    @Override
    public DocumentList<DocumentModel> get(ContentStore db, TemplateModel model, String key) {
        // document types are pluralized in model, so unpluralize
        try {
            String type = DocumentTypeUtils.unpluralize(key);
            return db.getAllContent(type);
        } catch (UnsupportedOperationException e) {
            return new DocumentList<>();
        }
    }

}

File: jbake-core/src/main/java/org/jbake/template/model/TypedDocumentsExtractor.kt
package org.jbake.template.model

import org.jbake.app.DocumentList
import org.jbake.model.DocumentTypeUtils
import org.jbake.template.TypedModelExtractor

class TypedDocumentsExtractor : TypedModelExtractor<DocumentList<*>> {

    override fun extract(context: RenderContext, key: String): DocumentList<*> {
        try {
            val type = DocumentTypeUtils.unpluralize(key)
            return context.db.getAllContent(type)
        } catch (e: UnsupportedOperationException) {
            return DocumentList<Any>()
        }
    }
}
